lib/evse_security/crypto/openssl/openssl_supplier.cpp

```diff
@@ -53,6 +77,235 @@ static CertificateValidationError to_certificate_error(const int ec) {
     }
 }
 
+const char* OpenSSLSupplier::get_supplier_name() {
+    return OPENSSL_VERSION_TEXT;
+}
+
+bool OpenSSLSupplier::supports_tpm() {
+    static bool support_checked = false;
+    static bool supports_tpm = false;
+
+    // TODO (ioan): Check if somehow the TPM provider was already
+    // loaded case in which we just need to do the self-test
+    if (support_checked == false) {
+        OSSL_PROVIDER* tpm2_provider = OSSL_PROVIDER_load(nullptr, PROVIDER_TPM);
+
+        if (tpm2_provider != nullptr) {
+            supports_tpm = OSSL_PROVIDER_available(nullptr, PROVIDER_TPM) && OSSL_PROVIDER_self_test(tpm2_provider);
+            OSSL_PROVIDER_unload(tpm2_provider);
+        } else {
+            supports_tpm = false;
+        }
+    }
+
+    return supports_tpm;
+}
+
+bool OpenSSLSupplier::supports_tpm_key_creation() {
+    if (supports_tpm()) {
+        return true;
+    }
+
+    return false;
+}
+
+static bool export_key_internal(const KeyGenerationInfo& key_info, const EVP_PKEY_ptr& evp_key) {
+    // write private key to file
+    if (key_info.private_key_file.has_value()) {
+        BIO_ptr key_bio(BIO_new_file(key_info.private_key_file.value().c_str(), "w"));
+
+        if (!key_bio) {
+            EVLOG_error << "Failed to create private key file!";
+            return false;
+        }
+
+        int success;
+        if (key_info.private_key_pass.has_value()) {
+            success = PEM_write_bio_PrivateKey(key_bio.get(), evp_key.get(), EVP_aes_128_cbc(), NULL, 0, NULL,
+                                               (void*)key_info.private_key_pass.value().c_str());
+        } else {
+            success = PEM_write_bio_PrivateKey(key_bio.get(), evp_key.get(), NULL, NULL, 0, NULL, NULL);
+        }
+
+        if (false == success) {
```

Comment:
success is int, so shouldn't this check be:
```suggestion
        if (0 == success) {
```

Reply:
Although using false here is a bit more clear in meaning...